app/services/DocumentService.ts

```
@@ -0,0 +1,201 @@
+        docCount++;
+      }
+      index++;
+    }
+
+    /** Sends list of Document IDs to JSON Body */
+    return documentList;
+  },
+
+  /**
+   * Takes the id from the request and looks for it in the
+   * dictionary object
+   * If the document is not there, sends a '404 Not Found' status.
+   * If the document is found, sends id, content, and metadata to Body
+   * @param req Express Request
+   * @param res Express Response
+   * @throws 'DocumentNotFoundException' when the document id given does not exist
+   */
+  async singleDoc(id) {
+    const docsObj = getFrontmatter();
+
+    /** Finds document in database using ID
+     *  Move this to controller once mark.ts
+     * is set up right
+     */
+    const document = docsObj[id];
+    
+      //document not found
+      if (!document)
+        throw new NotFoundException();
+      else {
+        //document found
+        return {
+          id: id,
+          content: document.content,
+          metadata: {
+            title: document.title,
+            description: document.description,
+            author: document.author,
+            group: document.group,
+            dateCreated: document.createdDate,
+            dateUpdated: document.updatedDate,
+          },
+        };
+      }
+  },
+};
+
+/** MARKDOWN FRONTMATTER PARSING */
+
+/** Holds frontmatter and markdown from the .md files */
+const docs = new Object();
+
+/**
+ * Parses Frontmatter and stores in dictionary
+ * @returns dictionary object
+ */
+export function parseFrontmatter() {
+  /** All filenames in directory /docs */
+  const filenames = fs.readdirSync("docs");
+
+  /** Each document is tokenized and variables in the frontmatter are read and stored in database */
+  filenames.forEach((file) => {
+    if (file.endsWith(".md")) {
+      //reads content of file and grabs the data from frontmatter
+      fs.readFile("docs/" + file, "utf8", (err, data) => {
+        if (err) {
+          console.error(err);
+          return;
```

Not sure what this is for. Might want to throw the error instead of returning nothing